File: AOJ/aoj0506.py
def arrange(s):
    m = len(s)
    for i in xrange(m):
        cnt = 0
        for j in xrange(m):
            if s[j] == s[i]:
                cnt += 1
            else:
                break

    return 

def solve(n, s):
    for i in xrange(n):
        s = arrange(s)

    print s

while 1:
    n = int(raw_input())
    if n == 0: break

    s = raw_input()
    solve(n, s)

File: AOJ/aoj0506.cpp
#include <iostream>
#include <string>
using namespace std;

#define FOR(i,a,b) for(int i=(a);i<(int)(b);i++)
#define rep(i,n) FOR(i,0,n)

string arrange(string s) {
    int m = s.size();
    string t = "";
    rep(i,m) {
        int j, cnt = 0;
        for(j=i; j<m and s[j]==s[i]; j++) cnt++;
        t += to_string(cnt) + s[i];
        i = j-1;
    }
    return t;
}

int main() {
    int n;
    while(cin>>n && n) {
        string s;
        cin >> s;
        rep(i,n) s = arrange(s);
        cout << s << endl;
    }

    return 0;
}
